packages/actor-init-sparql/lib/HttpServiceSparqlEndpoint.ts

```diff
@@ -158,8 +159,48 @@ Options:
   public async handleRequest(engine: ActorInitSparql, variants: { type: string, quality: number }[],
                              stdout: Writable, stderr: Writable,
                              request: http.IncomingMessage, response: http.ServerResponse) {
-    const mediaType: string = request.headers.accept && request.headers.accept !== '*/*'
-      ? require('negotiate').choose(variants, request)[0].type : null;
+
+    // Negotiate the best mediatype format
+    let negotiation = undefined;
+    let mediaType = null;
+    try {
+
+      negotiation = require('negotiate').choose(variants, request);
+      const isValid = request.headers.accept && request.headers.accept !== '*/*' && negotiation;
+      mediaType = isValid ? negotiation[0].type : null;
+
+      if (isValid && negotiation.length > 1) {
+        // The preferred negotiation
+        let finalNegotiation = negotiation[0];
+        // Best quality achievable
+        const quality = negotiation[0].q;
+        let found = false;
+        let k = 0;
+        // Iterate through all the variants in order
+        while (!found) {
+          const variant = variants[k];
+          let foundNegotiation = false;
+          let l = 0;
+          // Iterate through all the negotiation
+          while (!foundNegotiation) {
+            const negotiationElement = negotiation[l];
+            if(negotiationElement.type === variant.type){
+              foundNegotiation = true;
+              if(negotiationElement.q === quality){
+                found = true;
+                finalNegotiation = negotiationElement;
+              }
+            }
+            l++;
+          }
+          k++;
+        }
+        mediaType = finalNegotiation.type;
+      }
+
+    } catch (exception) {
```

Comment:
Why do you need a try-catch here?

Reply:
require('negotiate').choose(variants, request); Is not always possible since it throws an error so then i catch it but don't do anything with it since it will basically result into null and undefined of the previous declared values

Reply:
When does it throw an error?
We did not catch those errors before. Does this mean that we had a bug?

Reply:
The tests indeed never failed on it before. But somehow when I was writing it failed on the variant can not sort, so an issue in the negotiate package itself. I will recheck to look whether or not i did something fishy